macros/models/_project/chainlink/chainlink_ccip_macro.sql

```diff
@@ -0,0 +1,46 @@
+{% macro 
+    chainlink_ccip_tokens_transferred(
+        blockchain
+    ) 
+%}
+
+SELECT DISTINCT
+    le.blockchain,
+    le.block_time,
+    le.tx_hash,
+    COALESCE(le.total_tokens / POWER(
+        CAST(10 AS double precision),
+        CAST(t.decimals AS double precision)
+    ), 0) AS total_tokens,
+    COALESCE(sr.destination_blockchain, 'UNKNOWN') AS destination_blockchain,
+    te.contract_address AS token,
+    COALESCE(t.symbol, 'UNKNOWN') AS token_symbol,
+    COALESCE(p.price * (
+        le.total_tokens / POWER(
+            CAST(10 AS double precision),
+            CAST(t.decimals AS double precision)
+        )
+    ), 0) AS token_price
+FROM
+    {{ref('chainlink_' ~ blockchain ~ '_ccip_tokens_transferred_logs')}} le
+    INNER JOIN {{ref('chainlink_' ~ blockchain ~ '_ccip_send_requested')}} sr ON le.tx_hash = sr.tx_hash
+    {% if is_incremental() %}
+        WHERE {{ incremental_predicate('block_time') }}
+    {% endif %}
+    INNER JOIN  {{ source('erc20_' ~ blockchain, 'evt_transfer') }} te ON le.contract_address = te.to
+    AND te.evt_tx_hash = le.tx_hash
+    {% if is_incremental() %}
+        WHERE {{ incremental_predicate('block_time') }}
+    {% endif %}
+    LEFT JOIN {{ source('tokens', 'erc20') }} t ON te.contract_address = t.contract_address
+    {% if is_incremental() %}
+        AND {{ incremental_predicate('block_time') }}
+    {% endif %}
```

Comment:
```suggestion

```

Reply:
token metadata doesn't need to be filtered incrementally, it's just one row per address/chain